### ISSUE TITLE
Keras model.train() should automatically run table initializers.

### ISSUE THREAD

**Issue by yzhuang** [tensorflow/tensorflow#36348]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):  Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Centos 7, OS X 10.15
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.15 and 2.1
- Python version: 2.7 and 3.6
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**
We have Keras layers that use HashTables (e.g. `tf.lookup.StaticHashTable`), and these tables use initializers such as `tf.lookup.KeyValueTensorInitializer`.  When we perform model training using `model.train()` in **non-eager mode**, it does not run these table initializers and hence causes the training to crash.  Currently, we work around the issue with an ugly hack like this, by saving a reference to the initializer and running it manually. 
```
    if not tf.executing_eagerly():
      tf1.keras.backend.get_session().run(self.table._init_op)
```

**Describe the expected behavior**
Calling model.train() should initialize all initializers, including hash table initializers such as `tf.lookup.KeyValueTensorInitializer`.

**Code to reproduce the issue**
Any Keras layer using StaticHashTable would repro the problem. See
https://gist.github.com/yzhuang/0744b487c7a5ab1b65a5b152a06cda7c#file-keraslayertableinitialization-ipynb

**Suggestions**
My suggestion is to support HashTable initialization or publish documentation / guidance on how to use HashTable with Keras Layers.

Thanks! 🙏

**Comment by amahendrakar** [comment]
@yzhuang,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here. Thanks!

**Comment by yzhuang** [comment]
Hi @amahendrakar 

Here's a simple colab reproducing this issue on TF 1.15
https://gist.github.com/yzhuang/0744b487c7a5ab1b65a5b152a06cda7c#file-keraslayertableinitialization-ipynb

This colab shows that the eager mode correctly initializes the table, but non-eager mode does not initialize the table, and Keras Model API does not provide an API for table init ops.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF 1.15. Please find the Gist [here](https://colab.sandbox.google.com/gist/amahendrakar/c41e632bf014b78b8747500ab6d561ba/36348.ipynb). Thanks!

**Comment by tanzhenyu** [comment]
You can run tf.tables_initailizer, Keras historically doesn't have tables so it doesn't track that.
Moving forward, since we don't have any immediate plans to release 1.16 or backporting this, I think this might be the only way.

**Comment by yzhuang** [comment]
Hi @tanzhenyu 

This is not just affecting 1.15.  I have also tried 2.1, and the same colab can reproduce the issue in 2.1 as well by calling `tf.compat.v1.disable_eager_execution()`. 

Are there plans to fix this in 2.1? So it is not specific to 1.15.

"You can run tf.tables_initailizer".  In 2.1, we also no longer have access to the underlying sessions, any suggestions on how to run the initializer in TF 2.1?  Thanks.

**Comment by tanzhenyu** [comment]
> Hi @tanzhenyu
> 
> This is not just affecting 1.15. I have also tried 2.1, and the same colab can reproduce the issue in 2.1 as well by calling `tf.compat.v1.disable_eager_execution()`.
> 
> Are there plans to fix this in 2.x, since it is not specific to 1.15.
> 
> "You can run tf.tables_initailizer". In 2.x, we also no longer have access to the underlying sessions, any suggestions on how to run the initializer in TF 2.1? Thanks.

If you use tf.compat.v1.disable_eager_execution(), everything runs in graph mode, so you need to use tf.compat.v1.Session() to initialize it.
